File: lib/Models/Favorite.dart
class Favorite {
  int id;
  String accountId; // UUID của Account
  int songId; // ID của Song

  Favorite({
    required this.id,
    required this.accountId,
    required this.songId,
  });

  factory Favorite.fromJson(Map<String, dynamic> json) {
    return Favorite(
      id: json['id'],
      accountId: json['accountId'],
      songId: json['songId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'accountId': accountId,
      'songId': songId,
    };
  }
}

File: src/main/java/app_music/demo/Model/Favorite.java
package app_music.demo.Model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Favorite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "account_id")
    private Account account; // Người dùng

    @ManyToOne
    @JoinColumn(name = "song_id")
    private Song song; // Bài hát yêu thích
}
